fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java

```diff
@@ -74,11 +74,11 @@ public synchronized void createCatalog(CreateCatalogStmt stmt) throws DdlExcepti
         writeLock();
         try {
             catalogs.put(catalogName, catalog);
+            GlobalStateMgr.getCurrentState().getEditLog().logCreateCatalog(catalog);
```

Comment:
I think it's intended to do so in order to pass the jepsen test, the lock is mainly to block the `catalogs` reader. Before the log is synced to follower, read on leader cannot see the newly added catalog either, so that jepsen test will see a consistent state on leader and follower.

But actually under current design, we can only support eventual consistency, no causal consistency mechanism is provided, so the situation described in the issue is normal under eventual consistency. So the key point is that using a write lock here to log metadata change to just pass a specific jepsen test case is unnecessary. We need a fundamental  change in order to support causality which i think need further discuss wether we should support it.

Reply:
Causality is not necessary in our system. But the dirty read should be avoid. Add a write lock is to avoid the dirty read.

Reply:
Actually I don't dirty read should be avoid in this case, since catalog is ok to be eventual consistency, doesn't have to be linear consistency